channeldb/payments.go

```diff
@@ -570,7 +578,21 @@ type PaymentsResponse struct {
 // to a subset of payments by the payments query, containing an offset
 // index and a maximum number of returned payments.
 func (d *DB) QueryPayments(query PaymentsQuery) (PaymentsResponse, error) {
-	var resp PaymentsResponse
+	var (
+		resp           PaymentsResponse
+		minCreationSet = !query.MinCreationTimeNs.IsZero()
+		maxCreationSet = !query.MaxCreationTimeNs.IsZero()
+	)
+
+	// If MinCreationTimeNs filter is greater than MaxCreationTimeNs, we
+	// know there will be no matching results and can safely return early.
+	if minCreationSet && maxCreationSet {
+		if query.MinCreationTimeNs.After(query.MaxCreationTimeNs) {
```

Comment:
I don't think this will catch the case where min=max, and you wouldn't have any results either.